src/mobot_client/item_drop_session.py

```diff
@@ -441,68 +319,53 @@ def handle_item_drop_session_allow_contact_requested(self, message, drop_session
-            drop_session.state = ItemSessionState.COMPLETED
-            drop_session.save()
-            return
-
-        if message.text.lower() == 'p' or message.text.lower() == "privacy" or message.text.lower() == "privacy policy":
-            privacy_policy_url = drop_session.drop.store.privacy_policy_url
+            drop_session.state = SessionState.COMPLETED
+        elif message.text.lower() == 'p' or message.text.lower() == "privacy" or message.text.lower() == "privacy policy":
+            self.handle_privacy_policy_request(message, drop_session)
+        else:
             self.messenger.log_and_send_message(
-                drop_session.customer,
-                message.source,
-                ChatStrings.PRIVACY_POLICY_REPROMPT.format(url=privacy_policy_url),
+                drop_session.customer, message.source, ChatStrings.HELP
             )
-            return
+        drop_session.save()
 
+    def request_payment_from_customer(self, drop_session: DropSession, price_in_mob: Decimal):
         self.messenger.log_and_send_message(
-            drop_session.customer, message.source, ChatStrings.HELP
+            drop_session.customer,
+            drop_session.customer.phone_number.as_e164,
+            ChatStrings.PAYMENT_REQUEST.format(price=price_in_mob.normalize()),
         )
+        drop_session.state = SessionState.WAITING_FOR_PAYMENT
 
-    def handle_active_item_drop_session(self, message, drop_session):
-        print(drop_session.state)
-        if drop_session.state == ItemSessionState.WAITING_FOR_PAYMENT:
-            self.handle_item_drop_session_waiting_for_payment(message, drop_session)
-            return
-
-        if drop_session.state == ItemSessionState.WAITING_FOR_SIZE:
-            self.handle_item_drop_session_waiting_for_size(message, drop_session)
-            return
-
-        if drop_session.state == ItemSessionState.WAITING_FOR_ADDRESS:
-            self.handle_item_drop_session_waiting_for_address(message, drop_session)
-            return
-
-        if drop_session.state == ItemSessionState.WAITING_FOR_NAME:
-            self.handle_item_drop_session_waiting_for_name(message, drop_session)
-            return
-
-        if drop_session.state == ItemSessionState.SHIPPING_INFO_CONFIRMATION:
-            self.handle_item_drop_session_shipping_confirmation(message, drop_session)
-            return
-
-        if drop_session.state == ItemSessionState.ALLOW_CONTACT_REQUESTED:
-            self.handle_item_drop_session_allow_contact_requested(message, drop_session)
-            return
-
-    def handle_no_active_item_drop_session(self, customer, message, drop):
-        if not customer.phone_number.startswith(drop.number_restriction):
+    def show_catalog(self, message, drop_session: DropSession, message_to_send: str = ""):
+        drop = drop_session.drop
+        customer = drop_session.customer
+        available_options = drop.item.skus.all()
+        if available_options.count() == 0:
```

Comment:
The Django way of doing this check is using `.exists()`: https://docs.djangoproject.com/en/dev/ref/models/querysets/#exists

Reply:
Fixed